WD-Firmware/psp/api_inc/psp_macro_asm.S

```diff
@@ -0,0 +1,184 @@
+     *    is the industry''s de facto standard.                               *
+     *                                                                       *
+     *    Help yourself get started quickly while simultaneously helping     *
+     *    to support the FreeRTOS project by purchasing a FreeRTOS           *
+     *    tutorial book, reference manual, or both:                          *
+     *    http://www.FreeRTOS.org/Documentation                              *
+     *                                                                       *
+    ***************************************************************************
+
+    http://www.FreeRTOS.org/FAQHelp.html - Having a problem?  Start by reading
+    the FAQ page "My application does not run, what could be wrong?".  Have you
+    defined configASSERT()?
+
+    http://www.FreeRTOS.org/support - In return for receiving this top quality
+    embedded software for free we request you assist our global community by
+    participating in the support forum.
+
+    http://www.FreeRTOS.org/training - Investing in training allows your team to
+    be as productive as possible as early as possible.  Now you can receive
+    FreeRTOS training directly from Richard Barry, CEO of Real Time Engineers
+    Ltd, and the world's leading authority on the world's leading RTOS.
+
+    http://www.FreeRTOS.org/plus - A selection of FreeRTOS ecosystem products,
+    including FreeRTOS+Trace - an indispensable productivity tool, a DOS
+    compatible FAT file system, and our tiny thread aware UDP/IP stack.
+
+    http://www.FreeRTOS.org/labs - Where new FreeRTOS products go to incubate.
+    Come and try FreeRTOS+TCP, our new open source TCP/IP stack for FreeRTOS.
+
+    http://www.OpenRTOS.com - Real Time Engineers ltd. license FreeRTOS to High
+    Integrity Systems ltd. to sell under the OpenRTOS brand.  Low cost OpenRTOS
+    licenses offer ticketed support, indemnification and commercial middleware.
+
+    http://www.SafeRTOS.com - High Integrity Systems also provide a safety
+    engineered and independently SIL3 certified version for use in safety and
+    mission critical applications that require provable dependability.
+
+    1 tab == 4 spaces!
+*/
+#if __riscv_xlen == 64
+#define STORE    sd
+#define LOAD     ld
+#define LOG_REGBYTES 3
+#define ADDI     addiw
+#else
+#define STORE    sw
+#define LOAD     lw
+#define LOG_REGBYTES 2
+#define ADDI     addi
+#endif
+#define REGBYTES (1 << LOG_REGBYTES)
+
+.macro CALL_INT_HANDLER fptIntHandler
+    /* load the address of provided fptIntHandler */
+    la t0, \fptIntHandler
+    /* load the actual handler address */
+    LOAD t0, 0x0(t0)
+    /* invoker the interrupt handler */
+    jalr t0
+.endm
+
+/* Macro for saving task context */
+.macro pushREGFILE
+	/* make room in stack */
+	addi	sp, sp, -REGBYTES * 34
+	/* Save Context */
+	STORE	x1, 1 * REGBYTES(sp)
+	STORE	x2, 2 * REGBYTES(sp)
+	//STORE	x3, 3 * REGBYTES(sp)
```

Comment:
what happen to X3 (Global pointer) ?

Reply:
x3 is the gp - no need to save/restore gp as it doesn't change